### ISSUE TITLE
Define Model Serialization Data

### ISSUE THREAD

**Issue by ryanleary** [ryanleary/patter#1]
Serialized models must contain model metadata in addition to parameter weights.

**Parameters**
- *Required:* model parameter weights
- *Optional:* optimizer state

**Model Architecture**
- sample rate
- window size
- convolutional configuration
  - number of layers
  - stride
  - kernel size
  - activation
- rnn configuration
  - number of layers
  - rnn type
  - rnn size
  - activation
  - bidirectional

**Training Configuration**
- name/id of this model/experiment
- training set(s)/validation set(s) name/sha/identifier or 
- labels
- batch size
- optimizer
  - optimizer configuration (lr / annealing / momentum etc)
- max norm
- noise/augmentation configuration
  - tbd

**Training/Model Status**
- epoch times
- WER/CER/Loss @ each epoch
- best epoch (shall match weights, but may not be current epoch)
- WER/CER of model

**Comment by ryanleary** [comment]
Done.
